frontend/projects/upgrade/src/app/core/experiments/store/experiments.effects.ts

```diff
@@ -440,15 +441,48 @@ export class ExperimentEffects {
     )
   );
 
+  importExperiment$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(experimentAction.actionImportExperiment),
+      map(action => ({ experiments: action.experiments })),
+      filter(( {experiments} ) => !!experiments),
+      switchMap(({ experiments }) =>
+        this.experimentDataService.importExperiment(experiments).pipe(
+          switchMap((data: Experiment[]) => {
+            const experimentIds = data.map(exp => exp.id);
+            return [
+              experimentAction.actionImportExperimentSuccess(),
+              experimentAction.actionGetExperimentsSuccess({ experiments: data, totalExperiments: data.length }),
+              experimentAction.actionFetchExperimentStats({ experimentIds })
+            ];
+          }),
+          catchError(() => [experimentAction.actionImportExperimentFailure()])
+        )
+      )
+    )
+  );
+
   exportExperimentDesign$ = createEffect(() =>
     this.actions$.pipe(
       ofType(experimentAction.actionExportExperimentDesign),
-      map(action => ({ experimentId: action.experimentId })),
-      filter(( {experimentId} ) => !!experimentId),
-      switchMap(({ experimentId }) =>
-        this.experimentDataService.exportExperimentDesign(experimentId).pipe(
+      map(action => ({ experimentIds: action.experimentIds })),
+      filter(( {experimentIds} ) => !!experimentIds),
+      switchMap(({ experimentIds }) =>
+        this.experimentDataService.exportExperimentDesign(experimentIds).pipe(
           map((data: any) => {
-            this.download(data.name+'.json', data);
+            if (data.length > 1) {
+              var zip = new JSZip();
+              let index = 1;
+              data.forEach(experiment => {
+                zip.file(experiment.name+' (File '+index+').json', JSON.stringify(experiment));
+                index++;
+              });
+              zip.generateAsync({type:"base64"}).then(function (content) {
```

Comment:
I believe you should be able to use the normal fat-arrow function here? I don't think you should have to do the `bind(this)` thing.

```typescript
zip.generateAsync({type:"base64"}).then((content) => {
    this.download('Experiments.zip', content, true)
});
```

Reply:
Updated the code